### PR TITLE
[SDKS-7030] Update FeatureFlagWorker to add in the queue the ffnotifications

### PR DESCRIPTION
**Pull request by nmayorsplit**

### CODE REVIEW

File: client/src/main/java/io/split/storages/pluggable/adapters/UserCustomSplitAdapterProducer.java [hunk 1]
```diff
@@ -98,6 +99,13 @@ public void decreaseTrafficType(String trafficType) {
         }
     }
 
+    @Override
+    public void updateFeatureFlag(ParsedSplit parsedSplit) {
+        List<ParsedSplit> parsedSplits = new ArrayList<>();
+        parsedSplits.add(parsedSplit);
+        putMany(parsedSplits);
```
Comment (gthea) [inline client/src/main/java/io/split/storages/pluggable/adapters/UserCustomSplitAdapterProducer.java:106]: Not sure if you have `Collections` but if it's available..
```suggestion
        putMany(Collections.singletonList(parsedSplit));
```

File: client/src/main/java/io/split/engine/common/SynchronizerImp.java [hunk 2]
```diff
@@ -134,9 +138,16 @@ private void logCdnHeaders(String prefix, int maxRetries, int remainingAttempts,
...
+    public void refreshSplits(FeatureFlagChangeNotification featureFlagChangeNotification) {
 
-        if (targetChangeNumber <= _splitCacheProducer.getChangeNumber()) {
+        if (featureFlagChangeNotification.getChangeNumber() <= _splitCacheProducer.getChangeNumber()) {
+            return;
+        }
+
+        if (featureFlagChangeNotification.getFeatureFlagDefinition() != null &&
+            featureFlagChangeNotification.getPreviousChangeNumber() == _splitCacheProducer.getChangeNumber()){
+            SplitParser splitParser = new SplitParser();
```
[the remark this answers is not in the document]
Comment (nmayorsplit) [reply client/src/main/java/io/split/engine/common/SynchronizerImp.java]: Now it's passing in the constructor. I need it to parse the feature flag.

File: client/src/main/java/io/split/engine/common/SynchronizerImp.java [hunk 3]
```diff
@@ -183,10 +194,11 @@ public void refreshSplits(Long targetChangeNumber) {
...
     @Override
-    public void localKillSplit(String featureFlagName, String defaultTreatment, long newChangeNumber) {
-        if (newChangeNumber > _splitCacheProducer.getChangeNumber()) {
-            _splitCacheProducer.kill(featureFlagName, defaultTreatment, newChangeNumber);
-            refreshSplits(newChangeNumber);
+    public void localKillSplit(SplitKillNotification splitKillNotification) {
+        if (splitKillNotification.getChangeNumber() > _splitCacheProducer.getChangeNumber()) {
+            _splitCacheProducer.kill(splitKillNotification.getSplitName(), splitKillNotification.getDefaultTreatment(), splitKillNotification.getChangeNumber());
+            refreshSplits(new FeatureFlagChangeNotification(new GenericNotificationData(splitKillNotification.getChangeNumber(), null,
+                    null, null, null, null, null, null, null, null, null)));
```
[the remark this answers is not in the document]
Comment (nmayorsplit) [reply client/src/main/java/io/split/engine/common/SynchronizerImp.java]: I created a builder.

File: client/src/main/java/io/split/storages/pluggable/adapters/UserCustomSplitAdapterProducer.java (hunk 1 above)
Comment (nmayorsplit) [reply client/src/main/java/io/split/storages/pluggable/adapters/UserCustomSplitAdapterProducer.java]: I updated.

File: client/src/main/java/io/split/engine/common/SynchronizerImp.java [hunk 4]
```diff
@@ -134,9 +141,15 @@ private void logCdnHeaders(String prefix, int maxRetries, int remainingAttempts,
     }
 
     @Override
-    public void refreshSplits(Long targetChangeNumber) {
+    public void refreshSplits(FeatureFlagChangeNotification featureFlagChangeNotification) {
 
-        if (targetChangeNumber <= _splitCacheProducer.getChangeNumber()) {
+        if (featureFlagChangeNotification.getChangeNumber() <= _splitCacheProducer.getChangeNumber()) {
```
Comment (sanzmauro) [inline client/src/main/java/io/split/engine/common/SynchronizerImp.java:146]: this code should be into the Worker

File: client/src/main/java/io/split/engine/sse/workers/FeatureFlagWorkerImp.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+    public FeatureFlagWorkerImp(Synchronizer synchronizer) {
+        super("Feature flags");
+        _synchronizer = checkNotNull(synchronizer);
+    }
+
+    @Override
+    public void kill(SplitKillNotification splitKillNotification) {
+        try {
+            _synchronizer.localKillSplit(splitKillNotification);
```
Comment (sanzmauro) [inline client/src/main/java/io/split/engine/sse/workers/FeatureFlagWorkerImp.java:20]: we need to check this together.

**Review by sanzmauro — CHANGES_REQUESTED** [review]
we need to check the implementation. i will ping you

File: client/src/main/java/io/split/engine/sse/workers/FeatureFlagWorkerImp.java (hunk 5 above)
Comment (nmayorsplit) [reply client/src/main/java/io/split/engine/sse/workers/FeatureFlagWorkerImp.java:29]: I updated.

File: client/src/main/java/io/split/engine/common/SynchronizerImp.java (hunk 4 above)
Comment (nmayorsplit) [reply client/src/main/java/io/split/engine/common/SynchronizerImp.java]: I moved it to the worker.

File: client/src/main/java/io/split/client/SplitFactoryImpl.java [hunk 6]
```diff
@@ -240,8 +240,10 @@ public SplitFactoryImpl(String apiToken, SplitClientConfig config) throws URISyn
                 _impressionsManager, _eventsTask, _telemetrySyncTask, _uniqueKeysTracker);
         SplitAPI splitAPI = SplitAPI.build(_httpclient, buildSSEdHttpClient(apiToken, config, _sdkMetadata));
 
+        SplitParser splitParser = new SplitParser();
```
Comment (sanzmauro) [inline client/src/main/java/io/split/client/SplitFactoryImpl.java:243]: please don't create a new instance of `SplitParser`. Try to reuse the same one that you are using in the `SplitFetcher`.

**Review by sanzmauro — APPROVED** [review]

**Pull request merged**